tests/foreman/longrun/test_oscap.py

```diff
@@ -42,670 +42,593 @@
+    ).create()
+    # Create activation keys for rhel6, rhel7 and rhel8.
+    for repo in repo_values:
+        activation_key = entities.ActivationKey(
+            name=repo.get('akname'), environment=request.env, organization=module_org
+        ).create()
+        # Setup org for a custom repo for RHEL6, RHEL7 and RHEL8.
+        setup_org_for_a_custom_repo(
+            {
+                'url': repo.get('repo'),
+                'organization-id': module_org.id,
+                'content-view-id': request.content_view.id,
+                'lifecycle-environment-id': request.env.id,
+                'activationkey-id': activation_key.id,
+            }
         )
-        smart_proxy.import_puppetclasses(environment=cls.puppet_env.name)
-        env = entities.LifecycleEnvironment(organization=org, name=gen_string('alpha')).create()
-        # Create content view
-        content_view = entities.ContentView(organization=org, name=gen_string('alpha')).create()
-        # Create activation keys for rhel6, rhel7 and rhel8.
-        for repo in repo_values:
-            activation_key = entities.ActivationKey(
-                name=repo.get('akname'), environment=env, organization=org
-            ).create()
-            # Setup org for a custom repo for RHEL6, RHEL7 and RHEL8.
-            setup_org_for_a_custom_repo(
-                {
-                    'url': repo.get('repo'),
-                    'organization-id': org.id,
-                    'content-view-id': content_view.id,
-                    'lifecycle-environment-id': env.id,
-                    'activationkey-id': activation_key.id,
-                }
-            )
 
-        for content in cls.rhel8_content, cls.rhel7_content, cls.rhel6_content:
-            content = Scapcontent.info({'title': content}, output_format='json')
-            organization_ids = [
-                content_org['id'] for content_org in content.get('organizations', [])
-            ]
-            organization_ids.append(org.id)
-            Scapcontent.update({'title': content['title'], 'organization-ids': organization_ids})
-
-        return {
-            'org_name': org.name,
-            'cv_name': content_view.name,
-            'sat6_hostname': settings.server.hostname,
-            'ak_name': {'rhel8': ak_name_8, 'rhel7': ak_name_7, 'rhel6': ak_name_6},
-            'env_name': env.name,
+    for content in rhel8_content, rhel7_content, rhel6_content:
+        content = Scapcontent.info({'title': content}, output_format='json')
+        organization_ids = [content_org['id'] for content_org in content.get('organizations', [])]
+        organization_ids.append(module_org.id)
+        Scapcontent.update({'title': content['title'], 'organization-ids': organization_ids})
+
+    return request
+
+
+@pytest.mark.tier4
+@pytest.mark.libvirt_content_host
+@pytest.mark.upgrade
+@pytest.mark.parametrize(
+    'rhel_repo, content, profile1, profile2, profile3, distro',
```

Comment:
instead `profile1, profile2, profile3` making as `dsr_profile, pci_profile, osp_profile` would help in code readability ?

this is non-blocker

Reply:
yeah, it would make code more readable but the profiles can change and I think it's better to keep it general. For example, currently profile3 name for rhel6 is `usgcb` and not `ospp` as it's for rhel7 and rhel8. So I can't use `osp_profile` because of this exception.